### ISSUE TITLE
DeleteObjects API times out for deleting object versions

### ISSUE THREAD

**Issue by balamurugana** [minio/minio#9866]
When running below `minio-java` test, server fails to return response and make the test timing out.

```java
import io.minio.*;
import io.minio.messages.*;
import java.io.*;
import java.util.*;

public class Check {
  public static void main(String[] args) throws Exception {
    MinioClient client = new MinioClient(args[0], args[1], args[2]);

    client.makeBucket(MakeBucketArgs.builder().bucket("bala-test-bucket2").build());
    client.enableVersioning(EnableVersioningArgs.builder().bucket("bala-test-bucket2").build());

    ObjectWriteResponse v1 =
        client.putObject(
            PutObjectArgs.builder().bucket("bala-test-bucket2").object("my-object").stream(
                    new ByteArrayInputStream(new byte[] {'a'}), 1, -1)
                .build());

    ObjectWriteResponse v2 =
        client.putObject(
            PutObjectArgs.builder().bucket("bala-test-bucket2").object("my-object").stream(
                    new ByteArrayInputStream(new byte[] {'b'}), 1, -1)
                .build());

    List<DeleteObject> objects = new LinkedList<>();
    objects.add(new DeleteObject("my-object", v1.versionId()));
    objects.add(new DeleteObject("my-object", v2.versionId()));

    System.out.println(
        "Removing my-object and it's versions [" + v1.versionId() + ", " + v2.versionId() + "]");

    for (Result<DeleteError> result :
        client.removeObjects(
            RemoveObjectsArgs.builder().bucket("bala-test-bucket2").objects(objects).build())) {
      DeleteError error = result.get();
      System.out.println("Error in deleting object " + error.objectName() + "; " + error.message());
    }

    client.removeBucket(RemoveBucketArgs.builder().bucket("bala-test-bucket2").build());
  }
}
```
```java
Removing my-object and it's versions [5149ee3a-0388-4dd6-887c-3cf669075e02, 2df77c34-2ab9-4bad-b802-92b76c094ba5]
Exception in thread "main" java.net.SocketTimeoutException: timeout
	at okio.Okio$4.newTimeoutException(Okio.java:232)
	at okio.AsyncTimeout.exit(AsyncTimeout.java:286)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:241)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:358)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:230)
	at okhttp3.internal.http1.Http1ExchangeCodec.readHeaderLine(Http1ExchangeCodec.java:242)
	at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.java:213)
	at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.java:115)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:43)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229)
	at okhttp3.RealCall.execute(RealCall.java:81)
	at io.minio.MinioClient.execute(MinioClient.java:1258)
	at io.minio.MinioClient.execute(MinioClient.java:1406)
	at io.minio.MinioClient.executePost(MinioClient.java:1574)
	at io.minio.MinioClient.deleteObjects(MinioClient.java:7629)
	at io.minio.MinioClient$1$1.populate(MinioClient.java:3320)
	at io.minio.MinioClient$1$1.hasNext(MinioClient.java:3352)
	at Check.main(Check.java:33)
Caused by: java.net.SocketException: Socket closed
	at java.net.SocketInputStream.read(SocketInputStream.java:204)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at okio.Okio$2.read(Okio.java:140)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
	... 27 more
```

The same test works fine with AWS S3.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`minio version`): Development build using commit id `72743d15902717bbf656e4b7159c5e894b8c5312`
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version: `rm -fr ~/tmp/ed{1..4}; MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio123 ./minio server ~/tmp/ed{1..4}`
* Operating System and version (`uname -a`):
* Link to your project:

**Comment by balamurugana** [comment]
`mc admin trace` output
```
[bala@localhost mc]$ ./mc admin trace -v local
localhost [REQUEST s3.PutBucket] 04:25:04.578
localhost PUT /bala-test-bucket2
localhost Host: localhost:9000
localhost Accept-Encoding: identity
localhost X-Amz-Date: 20200618T042504Z
localhost Authorization: AWS4-HMAC-SHA256 Credential=minio/20200618/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=ea4e080bdfb1100cfd72d7cc6de7f79f0e56a7a4427fbf0261310dce0c021b8e
localhost Connection: Keep-Alive
localhost Content-Length: 0
localhost Content-Md5: 1B2M2Y8AsgTpgAmY7PhCfg==
localhost Content-Type: application/octet-stream
localhost User-Agent: MinIO (amd64; amd64) minio-java/7.1.0-DEV
localhost X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
localhost 
localhost [RESPONSE] [04:25:04.578] [ Duration 15.596ms  ↑ 129 B  ↓ 236 B ]
localhost 200 OK
localhost X-Xss-Protection: 1; mode=block
localhost Accept-Ranges: bytes
localhost Content-Length: 0
localhost Content-Security-Policy: block-all-mixed-content
localhost Location: /bala-test-bucket2
localhost Server: MinIO/DEVELOPMENT.GOGET
localhost Vary: Origin
localhost X-Amz-Request-Id: 161988C0B87C422E
localhost 
localhost 
localhost [REQUEST s3.PutBucketVersioning] 04:25:04.696
localhost PUT /bala-test-bucket2?versioning=
localhost Host: localhost:9000
localhost Authorization: AWS4-HMAC-SHA256 Credential=minio/20200618/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=22d7f3f2185dda95c11336e1a351a840803dee0788ae50e87c75e0e0d2ff8d72
localhost User-Agent: MinIO (amd64; amd64) minio-java/7.1.0-DEV
localhost X-Amz-Date: 20200618T042504Z
localhost Accept-Encoding: identity
localhost Connection: Keep-Alive
localhost Content-Length: 123
localhost Content-Md5: QQFYoy/mRYV9PGZUfFi0Bw==                                                                                                    
localhost Content-Type: application/octet-stream                                                                                                   
localhost X-Amz-Content-Sha256: 482d4e510b5cdcf0bbf3a044e82832dc711f4f28757a6c06eb21758282a73ef7                                                   
localhost <VersioningConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Status>Enabled</Status></VersioningConfiguration>              
localhost [RESPONSE] [04:25:04.696] [ Duration 16.094ms  ↑ 252 B  ↓ 207 B ]                                                                        
localhost 200 OK                                                                                                                                   
localhost Accept-Ranges: bytes
localhost Content-Length: 0                                                                                                                        
localhost Content-Security-Policy: block-all-mixed-content                                                                                         
localhost Server: MinIO/DEVELOPMENT.GOGET                                                                                                          
localhost Vary: Origin                                                                                                                             
localhost X-Amz-Request-Id: 161988C0BF7FDE73                                                                                                       
localhost X-Xss-Protection: 1; mode=block                                                                                                          
localhost                                                                                                                                          
localhost 
localhost [REQUEST s3.PutObject] 04:25:04.742
localhost PUT /bala-test-bucket2/my-object
localhost Host: localhost:9000
localhost X-Amz-Content-Sha256: ca978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee48bb                                                   
localhost X-Amz-Date: 20200618T042504Z                                                                                                             
localhost Authorization: AWS4-HMAC-SHA256 Credential=minio/20200618/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=4a95436f138b0b3e17e176cfed7735cd6dd212f96582ea0e192bf16d96b88bb3                                             
localhost Content-Length: 1                                                                                                                        
localhost Content-Md5: DMF1ucDxtqgxw5niaXcmYQ==                                                                                                    
localhost User-Agent: MinIO (amd64; amd64) minio-java/7.1.0-DEV                                                                                    
localhost Accept-Encoding: identity                                                                                                                
localhost Connection: Keep-Alive                                                                                                                   
localhost Content-Type: application/octet-stream                                                                                                   
localhost <BODY>                                                                                                                                   
localhost [RESPONSE] [04:25:04.742] [ Duration 15.922ms  ↑ 130 B  ↓ 303 B ]                                                                        
localhost 200 OK                                                                                                                                   
localhost Accept-Ranges: bytes
localhost ETag: "0cc175b9c0f1b6a831c399e269772661"                                                                                                 
localhost X-Amz-Request-Id: 161988C0C23DBB53                                                                                                       
localhost X-Xss-Protection: 1; mode=block                                                                                                          
localhost x-amz-version-id: 7fc2826c-59fa-4fa7-90eb-d9da9d5f97a9                                                                                   
localhost Content-Length: 0                                                                                                                        
localhost Content-Security-Policy: block-all-mixed-content                                                                                         
localhost Server: MinIO/DEVELOPMENT.GOGET                                                                                                          
localhost Vary: Origin                                                                                                                             
localhost <BODY>                                                                                                                                   
localhost 
localhost [REQUEST s3.PutObject] 04:25:04.758
localhost PUT /bala-test-bucket2/my-object
localhost Host: localhost:9000
localhost Connection: Keep-Alive                                                                                                                   
localhost Content-Md5: kutf/uauL+w61xx3dTFXjw==                                                                                                    
localhost Content-Type: application/octet-stream                                                                                                   
localhost X-Amz-Content-Sha256: 3e23e8160039594a33894f6564e1b1348bbd7a0088d42c4acb73eeaed59c009d                                                   
localhost Accept-Encoding: identity                                                                                                                
localhost Authorization: AWS4-HMAC-SHA256 Credential=minio/20200618/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=aa306bf27e5c94834468576b679f4f5977cb08d802219cf6ec8753a671602ff2                                             
localhost Content-Length: 1                                                                                                                        
localhost User-Agent: MinIO (amd64; amd64) minio-java/7.1.0-DEV                                                                                    
localhost X-Amz-Date: 20200618T042504Z                                                                                                             
localhost <BODY>                                                                                                                                   
localhost [RESPONSE] [04:25:04.758] [ Duration 11.137ms  ↑ 130 B  ↓ 303 B ]                                                                        
localhost 200 OK                                                                                                                                   
localhost Vary: Origin
localhost X-Xss-Protection: 1; mode=block                                                                                                          
localhost ETag: "92eb5ffee6ae2fec3ad71c777531578f"                                                                                                 
localhost Server: MinIO/DEVELOPMENT.GOGET                                                                                                          
localhost Content-Security-Policy: block-all-mixed-content                                                                                         
localhost X-Amz-Request-Id: 161988C0C37E0D8B                                                                                                       
localhost x-amz-version-id: f4527748-146b-4c8b-8c38-f110097cbc35                                                                                   
localhost Accept-Ranges: bytes                                                                                                                     
localhost Content-Length: 0                                                                                                                        
localhost <BODY>                                                                                                                                   
localhost 
localhost [REQUEST s3.DeleteMultipleObjects] 04:30:04.775
localhost POST /bala-test-bucket2?delete=
localhost Host: localhost:9000
localhost Content-Type: application/octet-stream                                                                                                   
localhost User-Agent: MinIO (amd64; amd64) minio-java/7.1.0-DEV                                                                                    
localhost X-Amz-Content-Sha256: 8a84b4ef2668bb51913092e3c722be4ad48a536e51b08987262e1122e123f1ed                                                   
localhost Accept-Encoding: identity                                                                                                                
localhost Connection: Keep-Alive                                                                                                                   
localhost Content-Length: 277                                                                                                                      
localhost Content-Md5: vMYoUaETX8KaCNpUZdwBfg==                                                                                                    
localhost X-Amz-Date: 20200618T042504Z                                                                                                             
localhost Authorization: AWS4-HMAC-SHA256 Credential=minio/20200618/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=2776519e7e1e44cf4642678e9a563b3d5d914785674b42f7cdaee1ca56458c73                                             
localhost <Delete xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Quiet>false</Quiet><Object><Key>my-object</Key><VersionId>7fc2826c-59fa-4fa7-90eb-d9da9d5f97a9</VersionId></Object><Object><Key>my-object</Key><VersionId>f4527748-146b-4c8b-8c38-f110097cbc35</VersionId></Object></Delete>       
localhost [RESPONSE] [04:30:04.775] [ Duration 5m0.001444s  ↑ 406 B  ↓ 801 B ]                                                                     
localhost 200 OK                                                                                                                                   
localhost Content-Length: 562
localhost Content-Security-Policy: block-all-mixed-content                                                                                         
localhost Content-Type: application/xml                                                                                                            
localhost Server: MinIO/DEVELOPMENT.GOGET                                                                                                          
localhost Vary: Origin                                                                                                                             
localhost X-Amz-Request-Id: 161988C0C51E14A8                                                                                                       
localhost X-Xss-Protection: 1; mode=block                                                                                                          
localhost Accept-Ranges: bytes                                                                                                                     
localhost <?xml version="1.0" encoding="UTF-8"?>                                                                                                   
<DeleteResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Error><Code>RequestTimeout</Code><Message>A timeout occurred while trying to lock a resource, please reduce your request rate</Message><Key>my-object</Key><VersionId>7fc2826c-59fa-4fa7-90eb-d9da9d5f97a9</VersionId></Error><Error><Code>RequestTimeout</Code><Message>A timeout occurred while trying to lock a resource, please reduce your request rate</Message><Key>my-object</Key><VersionId>f4527748-146b-4c8b-8c38-f110097cbc35</VersionId></Error></DeleteResult>                                                                 
localhost 
```

**Comment by harshavardhana** [comment]
Fixed by #9867
